src/ol/typedefs.js

```diff
@@ -104,7 +104,7 @@ ol.CircleRenderOptions;
  * red, green, and blue should be integers in the range 0..255 inclusive.
  * alpha should be a float in the range 0..1 inclusive. If no alpha value is
  * given then `1` will be used.
- * @typedef {Array.<number>}
+ * @typedef {Array.<number>|Uint8Array|Uint8ClampedArray}
```

Comment:
The `ol.color.asArray()` method accepts an `ol.Color` argument and throws when called with a `Uint8Array` or a `Uint8ClampedArray`.  The type cast in that function keeps the compiler from complaining about it.  I'll add some failing tests to demonstrate the issue.